src/cli/src/install/mod.rs

```diff
@@ -122,12 +122,12 @@ pub fn install_bin<P: AsRef<Path>, B: AsRef<[u8]>>(bin_path: P, bytes: B) -> Res
     let bin_path = bin_path.as_ref();
 
     // Create directories to bin_path if they do not exist
-    if let Some(parent) = bin_path.parent() {
-        std::fs::create_dir_all(&parent)?;
-    }
+    let parent = bin_path.parent()
+        .ok_or_else(|| IoError::new(ErrorKind::NotFound, "parent directory not found"))?;
+    std::fs::create_dir_all(&parent)?;
 
     // Write bin to temporary file
-    let tmp_dir = tempdir::TempDir::new("fluvio")?;
+    let tmp_dir = tempdir::TempDir::new_in(parent, "fluvio")?;
```

Comment:
Should the temp dir be named something that indicated that it is a temp dir for updating, such as `update_tmp`?

Reply:
Possibly? I don't really have a preference. Perhaps we can do something like `fluvio-tmp` so people know it's fluvio-related even if their binary is somewhere else (e.g. `/usr/local/lib` or something)